services/QuillLMS/app/workers/teacher_notifications/send_notification_worker.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module TeacherNotifications
+  class SendNotificationWorker
+    include Sidekiq::Worker
+
+    def perform(activity_session_id)
+      @activity_session = find_activity_session(activity_session_id)
+      return unless @activity_session
+
+      TeacherNotification.transaction do
+        send_complete_diagnostic
+        send_complete_all_diagnostic_recommendations
+        send_complete_all_assigned_activities
+      end
+    end
+
+    private def find_activity_session(activity_session_id)
+      ::ActivitySession.where(id: activity_session_id)
+        .where.not(completed_at: nil)
+        .includes(:user)
+        .includes(teachers: :teacher_notification_settings)
+        .includes(:classroom)
+        .includes(activity: :classification)
+        .includes(unit_template: :recommendations)
+        .first
+    end
+
+    private def send_complete_diagnostic
```

Comment:
Note, every method in this file has `activity_session` as its base object. Take or leave, you could consider moving some methods there, e.g. `activity_session.send_complete_diagnostic_notification`